### PR TITLE
client/asset/dcr: retire splitFunds cache

### PR DESCRIPTION
**Pull request by itswisdomagain**
Issue with outputs spent in txs broadcast with sendrawtransaction being
temporarily returned by listunspent is [now fixed in dcrwallet](https://github.com/decred/dcrwallet/commit/f5bd1a27b888a1fc5df1efb5a83d2785a4a04fd3). Outputs
spent this way (such as in split transactions) are no longer cached in
the `splitFunds` map.

Min. required dcrwallet version bumped.

### CODE REVIEW

File: client/asset/dcr/dcr.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ const (
 )
 
 var (
-	requiredWalletVersion = dex.Semver{Major: 8, Minor: 3, Patch: 0}
+	requiredWalletVersion = dex.Semver{Major: 8, Minor: 4, Patch: 0}
```
Comment (itswisdomagain) [inline client/asset/dcr/dcr.go:56]: To test this PR against dcrwallet master, please undo this change. Making the change now so "we" don't forget to bump this version requirement before merging.

File: client/asset/dcr/dcr.go [hunk 2]
```diff
@@ -820,35 +818,11 @@ func (dcr *ExchangeWallet) split(value uint64, lots uint64, coins asset.Coins, i
...
-	// can be unlocked when the swap is sent. The helps to deal with a timing
-	// issue with dcrwallet where listunspent might still return outputs that
-	// were just spent in a transaction broadcast with sendrawtransaction.
-	// Instead, we'll keep them locked until the split output is spent.
-	dcr.splitFunds[op.pt] = fundingCoins
+	// Unlock the spent coins.
+	err = dcr.returnCoins(coins)
+	if err != nil {
+		dcr.log.Errorf("error returning coins spent in split transaction %v", coins)
+	}
```
Comment (itswisdomagain) [inline client/asset/dcr/dcr.go:825]: This reminds me that we have to explicitly unlock coins that we locked after they're spent otherwise listlockunspent might continue to return them.

dcr.sendRegFee and dcr.Withdraw both call dcr.fund to select input coins, dcr.fund locks the selected coins before returning them but dcr.sendRegFee and dcr.Withdraw do not unlock the coins after spending the coins.

First thought was make `dcr.fund` accept a bool parameter that determines whether to lock the selected coins - basically, don't lock if we're gonna be spending them right away but inbetween the brief moment between when the selected coins are returned and when they're spent, listunspent would return them, because the fundingMtx lock is released when dcr.fund returns. Also, another time when we should not lock the selected coins are when we plan to spend them in a split tx right away but we may end up not spending them (if it's uneconomical to use a split) and then we'd need to lock the coins.

The plenty rant above is to lay the groundwork for the changes coming in my next commit: don't lock coins in dcr.fund or even dcr.split. Let callers lock the fundingMtx before calling dcr.fund and dcr.split and only release the lock after either spending the selected coins or locking them.

Comment (itswisdomagain) [reply client/asset/dcr/dcr.go:825]: Okay, the only issue is here
> dcr.sendRegFee and dcr.Withdraw both call dcr.fund to select input coins, dcr.fund locks the selected coins before returning them but dcr.sendRegFee and dcr.Withdraw do not unlock the coins after spending the coins.

which does not seem to have been an issue before now.

Skipping this change for this PR. It's a better candidate for another PR I have in the works.

**Review by JoeGruffins — APPROVED** [review]

**Comment by chappjc** [comment]
Not a high priority, but requesting review from @buck54321 since the `splitFunds` map was your solution, so you are in the best position to evaluate this.

File: client/asset/dcr/dcr.go [hunk 3]
```diff
@@ -820,35 +818,11 @@ func (dcr *ExchangeWallet) split(value uint64, lots uint64, coins asset.Coins, i
 		dcr.log.Errorf("error locking funding coin from split transaction %s", op)
 	}
 
-	// NOTE: Can't return coins yet, because dcrwallet doesn't recognize them as
```
Comment (buck54321) [inline client/asset/dcr/dcr.go:823]: There's still one reference to `splitFunds` above here at line 799.

**Review by buck54321 — APPROVED** [review]
Looks good.

**Comment by buck54321** [comment]
Just documenting that the `splitFunds` cache is no longer needed because of @itswisdomagain's work in https://github.com/decred/dcrwallet/pull/1846.
